2-18-responsify

michael-duer opened the pull request.

UnibeRoger committed:
Incomplete Changes to be remembered

UnibeRoger committed:
Merge branch '2-17-DisplayImage' into 2-18-Responsify

UnibeRoger committed:
done

laurabuenemann commented on frontend/src/components/ImageViewerContainer/ImageViewerContainer.js:
```diff
@@ -137,6 +146,21 @@ class ImageViewerContainer extends React.Component {
     }
   }
 
```
Für diese beiden Methoden würde ich noch mehr Kommentare einfügen, die klar machen, welcher Teil genau dafür verantwortlich ist, dass das Bild die gleiche Grösse behält beim Verändern der Fenstergrösse. Wäre das zum Beispiel das Aufrufen von this.state.renderer.resetCamera() und this.vtkContext.renderWindow.render() in handleResize()?

michael-duer replied:
Für die beiden Funktionen sowie für weitere Teile im Component habe ich nun noch genauere Kommentare hinzugefügt.

laurabuenemann commented on frontend/src/App.js:
```diff
@@ -42,7 +42,7 @@ export default function App() {
             <Routes>
               <Route path="/" element={<MetaInformationContainer />} />
-              <Route path="/image-viewer" element={<ImageViewerContainer />} />
+              <Route path="/image-viewer" element={<div style = {{display: "flex",flexDirection:"row"}}><ImageViewerContainer /><ImageViewerContainer /></div>} />
```
Hier wäre ein Kommentar  toll. Vllt: 
// Renders the image viewer using flexbox layout with a row direction, allowing the ImageViewerContainer components to be displayed side by side horizontally

laurabuenemann commented on backend/src/dicoms/dicom_slice.js:
```diff
@@ -11,12 +11,13 @@ export const DicomSlice = async (req, res, models) => {
     console.log({ idSeries, position });
     if (IsInvalid(idSeries)) {
-      throw Error('Invalid Series or your are not authenticated');
+      throw Error('1Invalid Series or your are not authenticated');
```
Für was braucht es die 1?

laurabuenemann commented on frontend/src/App.js:
Kommentar hinzufügen, sonst topp

michael-duer replied:
Ich habe Kommentare hinzugefügt, so dass es nun klarer sein sollte was hier passiert.

laurabuenemann commented on frontend/src/components/ImageViewerContainer/ImageViewerContainer.js:
Hier wären etwas mehr Kommentare auch super.

michael-duer committed:
Refactor components to improve clarity and fix canvas overlap

- Remove unused code
- Enhance documentation with additional comments and correct typos
- Resolve canvas element overlap with parent div for consistent UI rendering

michael-duer committed:
Replace magic number with descriptive variable

UnibeRoger committed:
Merge branch 'main' into 2-18-Responsify
